### PR TITLE
De-nest finder-frontend styles

### PR DESCRIPTION
**Pull request by andysellick**
Continuing my mission to de-nest all the styles in finder-frontend from underneath a single id.

---

## Search page examples to sanity check:

- http://finder-frontend-pr-1163.herokuapp.com/search/research-and-statistics/email-signup
- http://finder-frontend-pr-1163.herokuapp.com/search/all
- http://finder-frontend-pr-1163.herokuapp.com/search/research-and-statistics
- http://finder-frontend-pr-1163.herokuapp.com/search/advanced?group=guidance_and_regulation&topic=%2Feducation
- http://finder-frontend-pr-1163.herokuapp.com/drug-device-alerts
- http://finder-frontend-pr-1163.herokuapp.com/find-eu-exit-guidance-business

[Other finders](https://live-stuff.herokuapp.com/finders)

Trello card: https://trello.com/c/geNECHkE/788-restructure-finder-frontend-css-to-not-all-sit-under-a-single-id

### CODE REVIEW

File: app/assets/stylesheets/finder_frontend.scss [hunk 1]
```diff
@@ -56,119 +56,86 @@
...
-    }
+  .js-required {
+    display: none;
+  }
 
-    p {
-      max-width: $full-width;
-    }
+  .govuk-select {
+    width: 100%;
```
Comment (injms) [inline app/assets/stylesheets/finder_frontend.scss]: Would it be better to use the `govuk-!-width-full` class on the element to get the same outcome?

File: app/assets/stylesheets/finder_frontend.scss [hunk 2]
```diff
@@ -56,119 +56,86 @@
...
-      border-left: solid 4px $govuk-error-colour;
-    }
+  @include media(tablet) {
+    max-width: $two-thirds;
   }
 
-  .search-results-title {
-    border-bottom: 1px solid govuk-colour('black');
-    padding-bottom: govuk-spacing(4);
+  fieldset.invalid {
```
Comment (injms) [inline app/assets/stylesheets/finder_frontend.scss]: Could using `.govuk-form-group--error` replace `fieldset.invalid`?

**Comment by vanitabarrett** [comment]
@andysellick Could you get this deployed to the Heroku review app so we can check the changes visually too, please?

File: app/assets/stylesheets/finder_frontend.scss (hunk 1 above)
Comment (andysellick) [reply app/assets/stylesheets/finder_frontend.scss]: I don't think we can use this here as it's applying this style to a component. Select components are the width they are, there's no GOVUK Frontend alternative. We could update the component to include this option, that might be worth doing in the future.

**Comment by andysellick** [comment]
@vanitabarrett hmm odd that it hasn't done it automatically. Afraid I don't have heroku access to deploy it (yeah, I know) any chance you could give it a kick?

File: app/assets/stylesheets/finder_frontend.scss (hunk 2 above)
Comment (andysellick) [reply app/assets/stylesheets/finder_frontend.scss]: Possibly. TBH I hadn't finished investigating this properly - decided it was better to stop here and get it reviewed while I'm not able to work on it anymore this week. Will make a note to investigate further for the next round.

**Comment by injms** [comment]
Probably another thing for another day - on the [email signup for drugs](https://finder-frontend-pr-1163.herokuapp.com/drug-device-alerts/email-signup) this is the default:

Screen Shot 2019-06-05 at 11 51 37

Should the parent checkbox also be selected?

File: app/views/email_alert_subscriptions/new.html.erb [hunk 3]
```diff
@@ -7,46 +7,41 @@
...
-
-      <%= render "govuk_publishing_components/components/button", {
-        text: "Create subscription",
-        margin_bottom: true
-      } %>
+<%= form_tag email_alert_subscriptions_path(subscriber_list_params: subscriber_list_params), class: 'signup-choices' do %>
+  <% if @signup_presenter.hidden_choices.any? %>
+    <%= render partial: 'govuk_publishing_components/components/title', locals: {
+      title: @signup_presenter.name,
+      context: "Email alert subscription"
```
Comment (vanitabarrett) [inline app/views/email_alert_subscriptions/new.html.erb]: I know this isn't your change, but is it worth putting this in a locale file?

File: app/views/email_alert_subscriptions/new.html.erb [hunk 4]
```diff
@@ -7,46 +7,41 @@
...
+    } %>
+    <% @signup_presenter.hidden_choices.each do |hidden_choice| %>
+      <%= hidden_field_tag hidden_choice[:name], hidden_choice[:value] %>
+    <% end %>
+  <% elsif @signup_presenter.can_modify_choices? %>
+    <%= render "govuk_publishing_components/components/checkboxes", {
+      name: nil,
+      heading: @signup_presenter.name,
+      hint_text: "Select the email alerts you need.",
```
Comment (vanitabarrett) [inline app/views/email_alert_subscriptions/new.html.erb]: Same comment here about the locale file

File: app/views/email_alert_subscriptions/new.html.erb [hunk 5]
```diff
@@ -7,46 +7,41 @@
...
+      items: @signup_presenter.choices_formatted,
+      is_page_heading: true,
+    } %>
+    <% if @error_message.present? %>
+      <p class="signup-choices__message"><%= @error_message %></p>
     <% end %>
+  <% else %>
+    <%= render partial: 'govuk_publishing_components/components/title', locals: {
+      title: @signup_presenter.name,
+      context: "Email alert subscription"
```
Comment (vanitabarrett) [inline app/views/email_alert_subscriptions/new.html.erb]: Same comment here about the locale file

File: app/views/email_alert_subscriptions/new.html.erb [hunk 6]
```diff
@@ -7,46 +7,41 @@
...
-  </div>
-</div>
+  <% if @signup_presenter.body %>
+    <div class="govspeak-wrapper">
+      <%= render 'govuk_publishing_components/components/govspeak', content: @signup_presenter.body.html_safe %>
+    </div>
+  <% end %>
+
+  <%= render "govuk_publishing_components/components/button", {
+    text: "Create subscription",
```
Comment (vanitabarrett) [inline app/views/email_alert_subscriptions/new.html.erb]: Same comment here about the locale file

File: app/assets/stylesheets/finder_frontend.scss [hunk 7]
```diff
@@ -71,18 +71,18 @@
   margin-bottom: govuk-spacing(6);
 }
 
+.logo {
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/assets/stylesheets/finder_frontend.scss]: Great spot, thanks. Have BEM-ified the class a bit to `finder-logo`.

File: app/views/finders/show.html.erb [hunk 8]
```diff
@@ -77,9 +77,11 @@
 
     <% if finder.related.any? %>
       <div class="related-links govuk-grid-column-one-third">
-        <ul class="finder-results">
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/views/finders/show.html.erb]: Another great spot, thanks. I've added the class back in but changed it to be more obviously js related (so no one else falls into the same trap I did). Appearance should be unaffected.

File: app/views/email_alert_subscriptions/new.html.erb (hunk 3 above)
Comment (andysellick) [reply app/views/email_alert_subscriptions/new.html.erb]: I agree, but I think that's a change for a separate PR. I think I've spotted more of these, could be a bit of a rabbit hole.

File: app/views/finders/show.html.erb (hunk 8 above)
[the remark this answers is not in the document]
Comment (andysellick) [reply app/views/finders/show.html.erb]: Oop, actually a bit more complicated. Just working on a fix now.

[the remark this answers is not in the document]
Comment (andysellick) [reply app/views/finders/show.html.erb]: I've finalised my changes. The `finder-results` class remains as it's being used by the main results lists. I've also added a `js-finder-results` class that is only used by the JS. It's slightly weird that this means this is nested inside itself (see `finders/search_results.html.erb`) but this was technically how it was before. At some point we should investigate exactly how this works and whether the js class needs to be nested like this, but this is outside the scope here.

**Pull request merged**